### ISSUE TITLE
Inconsistent use of operation vs operator throughout the text

### ISSUE THREAD

**Issue by devreal** [mpi-forum/mpi-issues#718]
# Problem

Stray comment by @RolfRabenseifner on https://github.com/mpi-forum/mpi-standard/pull/637:

>This merged https://github.com/mpi-forum/mpi-standard/pull/637 is not consistent with the remaining MPI.
>The closed and not merged https://github.com/mpi-forum/mpi-standard/pull/490 would have done this for collectives.tex (plus many additional changes).
>
> Both issues https://github.com/mpi-forum/mpi-issues/issues/429 and https://github.com/mpi-forum/mpi-issues/issues/280 have not helped to keep MPI consistent.
>
> We still have one or more PRs in the voting pipe that use the the wording "operation handle" as currently correct, because this wording is introduced in the section on MPI_REDUCE and not yet changed.
>
>How to proceed?

@Wee-Free-Scot  added:

> The problem is not with this PR, but with other locations within the MPI Standard that have not used the correct word for the intended concept.
>
> The changes in PR https://github.com/mpi-forum/mpi-standard/pull/490 are valuable and should be incorporated as soon as possible. Unfortunately, this PR was closed when MPI-4.0 was finished (along with all other PRs that targeted MPI-4.0, but missed the deadline for inclusion) and was never re-opened or pursued. This is a regrettable oversight, but not a reason to remove correct text fixes in a different chapter.
>
> @schulzm @wesbland can we do something within the procedural rules to expedite the changes in closed-not-merged PR 490? I'm thinking perhaps we could declare they are chapter committee level changes?
>
> @RolfRabenseifner do you know of any instances of operation/operator confusion outside the collective and RMA chapters?
>
> Some potential examples (each needs to be checked manually) from a non-exhaustive search:
> ```
> $ egrep -nRi "reduc.*operation" --include="*.tex" --exclude="coll.tex" .
> ./chap-binding/binding-2.tex:1579:committed; they can be used with predefined reduction operations.
> ./chap-binding/binding-2.tex:4533:delete functions are associated with attribute keys; reduce operations
> ./chap-binding/binding-2.tex:4572:\subsubsection{Reduce Operations}
> ./chap-binding/binding-2.tex:4573:\mpitermtitleindex{reduction operations}
> ./chap-binding/binding-2.tex:4581:Reduce operations receive as one of their arguments the datatype of the
> ./chap-changes/changes.tex:1929:\mpidtype{MPI\_CHAR} for printable characters is now defined for C type char (instead of signed char). This change should not have any impact on applications nor on \MPI/ libraries (except some comment lines), because printable characters could and can be stored in any of the C types char, signed char, and unsigned char, and \mpidtype{MPI\_CHAR} is not allowed for predefined reduction operations.
> ./chap-changes/changes.tex:1978:Predefined parameterized datatypes (e.g., returned by\\\mpifunc{MPI\_TYPE\_CREATE\_F90\_REAL}) and optional named predefined datatypes (e.g. \mpidtype{MPI\_REAL8}) have been added to the list of valid datatypes in reduction operations.
> ./chap-ei/ei-2.tex:403:This example shows the code for a user-defined reduce operation on an
> ./chap-one-side/one-side-2.tex:1656:reduction operations such as \mpifunc{MPI\_REDUCE}.
> ./chap-one-side/one-side-2.tex:1833:reduction operations, such as \mpifunc{MPI\_REDUCE} and others.
> ./func-index-clean.tex:637:  \item predefined reduction operations, \hyperindexformat{\bold}{214}
> ./func-index-clean.tex:723:  \item reduction operations, \hyperindexformat{\bold}{212},
> ./func-index-clean.tex:1011:  \item user-defined reduction operations,
> ./func-index-clean.tex:4436:\renewcommand{\@introtext}{This index lists the C typedef names for callback routines, such as those used with attribute caching or user-defined reduction operations.  Fortran example prototypes are given near the text of the C name.}
> ```

# Proposal

The respective chapter committees should address this.

# Changes to the Text

Use `operator` when referring `MPI_Op` and `operation` when referring to communication primitives.

MPI_SUM is an MPI operator, reduction is an MPI operation, MPI_REDUCE is an MPI procedure that performs an MPI reduction operation using an MPI operator.

# Impact on Implementations

None.

# Impact on Users

Consistency.

# References and Pull Requests

Fixed in collectives by PR: https://github.com/mpi-forum/mpi-standard/pull/837

RMA PR (merged): https://github.com/mpi-forum/mpi-standard/pull/637
RMA issue (closed): https://github.com/mpi-forum/mpi-issues/issues/429

Pre-MPI-4.1 issue for collectives chapter: https://github.com/mpi-forum/mpi-issues/issues/280
Never-merged PR for (mostly) collectives chapter: https://github.com/mpi-forum/mpi-standard/pull/490

**Comment by wesbland** [comment]
I agree that https://github.com/mpi-forum/mpi-standard/pull/490 is mostly small changes, but I'm not sure if we should declare it CC to avoid the voting requirements. All of the other changes of this type have, for better or worse, been treated as normative changes and gone through the full vote cycle.

I think there's also still a number of chapters who may not have finished the cleanup tasks that we have wanted to finish for MPI 4.1. So there's probably more example of inconsistency than just this one.

I'm not saying that this shouldn't be allowed to go in, but my preference would be to get a decision from the group at large rather than a declaration from @wgropp or me about the scope.

Speaking for myself, I'd probably agree that this could be considered a CC change as it doesn't seem like it involves the type of manipulations that we had to go through to fix things like with the term "process".

**Comment by wgropp** [comment]
I agree with @wesbland here - I think that we should get general approval, though as long as it is clarifications/corrections, I'd like to see it included in 4.1.

**Comment by RolfRabenseifner** [comment]
In my opinion, the best is to do an informal reading of this PR at the July meeting.
In principle, exactly such corrections of inconsistencies produced in the MPI-4.1 process is the goal of the upcoming review process. I tried to do these things already before the July meeting to get it done during the July meeting, and not at the September meeting., because I will not be available after the July meeting. 
And after this informal reading the chapter committees can do further corrections if needed and then vote the directly PR into MPI-4.1.

**Comment by wesbland** [comment]
I'll add it to the agenda.

**Comment by devreal** [comment]
@jaegerj are you planning to work on this for MPI 6? Right now the document is inconsistent with the RMA chapter having made the switch so it would be good to consistently use `operator` when referring to `MPI_Op`.

**Comment by jaegerj** [comment]
@devreal yes I will deal with it for MPI 6
